src/sidebar/services/api.js

```diff
@@ -39,31 +42,25 @@ function stripInternalProperties(obj) {
 }
 
 /**
+ * @template {object} Body
  * @typedef APIResponse
- * @prop {any} data -
+ * @prop {Body} data -
  *  The JSON response from the API call, unless this call returned a
  *  "204 No Content" status.
  * @prop {string|null} token - The access token that was used to make the call
  *   or `null` if unauthenticated.
  */
 
-/**
- * Options controlling how an API call is made or processed.
- *
- * @typedef APICallOptions
- * @prop {boolean} [includeMetadata] - If false (the default), the response is
- *   just the JSON response from the API. If true, the response is an `APIResponse`
- *   containing additional metadata about the request and response.
- */
-
 /**
  * Function which makes an API request.
  *
- * @callback APICallFunction
- * @param {Record<string, any>} params - A map of URL and query string parameters to include with the request.
- * @param {object} [data] - The body of the request.
- * @param {APICallOptions} [options]
- * @return {Promise<any|APIResponse>}
+ * @template {Record<string, any>} Params
+ * @template {object} Body
+ * @template Result
+ * @callback APICall
```

Comment:
Unfortunately the JSDoc syntax for defining generic types is rather verbose. Fortunately we don't need to use them all that often. Honestly I think the best way to address this would be better syntax highlighting. It's a bit better in VS Code than in GitHub. The comment is still parsed correctly with a new line between the `@template` and the rest, but I'm not sure it is safe to rely on that.